src/backend/access/external/pxffilters.c

```diff
@@ -851,12 +852,51 @@ append_attr_from_var(Var* var, List* attrs)
 	return attrs;
 }
 
+/*
+ * append_attr_from_func_args
+ *
+ * extracts all columns from FuncExpr into attrs
+ * assigns false to expressionIsSupported if at least one of items is not supported
+ */
+static List*
+append_attr_from_func_args(FuncExpr *expr, List* attrs, bool* expressionIsSupported) {
+	ListCell *lc = NULL;
+	if (!expressionIsSupported) {
```

Comment:
this can be the first check, so that you don't have to allocate lc variable if expression is not supported

Reply:
Sure, updated